contribs/application/src/main/java/org/matsim/application/analysis/population/TripAnalysis.java

```diff
@@ -188,6 +188,45 @@ public Integer call() throws Exception {
 		return 0;
 	}
 
+	private Character detectDelimiter(String path) throws IOException {
+		// Create a map to count occurrences of potential delimiters
+		Map<Character, Integer> delimiterCounts = new HashMap<>();
+
+		BufferedReader reader = null;
+		try {
+			if (path.endsWith(".gz")) {
+				reader = new BufferedReader(new InputStreamReader(new GZIPInputStream(new FileInputStream(path))));
+			} else if (path.endsWith(".csv")) {
+				reader = new BufferedReader(new InputStreamReader(new FileInputStream(path)));
+			} else {
+				log.error("Usupported file format.");
+			}
+
+			String firstLine = reader.readLine();
+
+			delimiterCounts.put(',', firstLine.split(",").length);
```

Comment:
I think you could use a util library, something like `StringUtils.countMatches`. The current solution creates unnecessary new strings